gobblin-yarn/src/main/java/gobblin/yarn/YarnHelixUtils.java

```diff
@@ -199,4 +206,48 @@ public static void addFileAsLocalResource(FileSystem fs, Path destFilePath, Loca
 
     return environmentVariableMap;
   }
-}
+
+  public static Optional<Integer> takePort(int desiredPort) {
```

Comment:
Not adamant on it, just curious. Makes sense for the the ports to be grouped together.